src/common/filesystembase.h

```diff
@@ -0,0 +1,189 @@
+    QString OCSYNC_EXPORT longWinPath(const QString &inpath);
+
+    /**
+     * @brief Checks whether a file exists.
+     *
+     * Use this over QFileInfo::exists() and QFile::exists() to avoid bugs with lnk
+     * files, see above.
+     */
+    bool OCSYNC_EXPORT fileExists(const QString &filename, const QFileInfo & = QFileInfo());
+
+    /**
+     * @brief Rename the file \a originFileName to \a destinationFileName.
+     *
+     * It behaves as QFile::rename() but handles .lnk files correctly on Windows.
+     */
+    bool OCSYNC_EXPORT rename(const QString &originFileName,
+        const QString &destinationFileName,
+        QString *errorString = NULL);
+
+    /**
+     * Rename the file \a originFileName to \a destinationFileName, and
+     * overwrite the destination if it already exists - without extra checks.
+     */
+    bool uncheckedRenameReplace(const QString &originFileName,
+        const QString &destinationFileName,
+        QString *errorString);
+
+    /**
+     * Removes a file.
+     *
+     * Equivalent to QFile::remove(), except on Windows, where it will also
+     * successfully remove read-only files.
+     */
+    bool OCSYNC_EXPORT remove(const QString &fileName, QString *errorString = 0);
+
+    /**
+     * Replacement for QFile::open(ReadOnly) followed by a seek().
+     * This version sets a more permissive sharing mode on Windows.
+     *
+     * Warning: The resulting file may have an empty fileName and be unsuitable for use
+     * with QFileInfo! Calling seek() on the QFile with >32bit signed values will fail!
+     */
+    bool openAndSeekFileSharedRead(QFile *file, QString *error, qint64 seek);
+
+#ifdef Q_OS_WIN
+    /**
+     * Returns the file system used at the given path.
+     */
+    QString fileSystemForPath(const QString &path);
+#endif
+
+    QByteArray OCSYNC_EXPORT calcMd5(const QString &fileName);
+    QByteArray OCSYNC_EXPORT calcSha1(const QString &fileName);
+#ifdef ZLIB_FOUND
+    QByteArray OCSYNC_EXPORT calcAdler32(const QString &fileName);
+#endif
+
+    /**
+     * Returns a file name based on \a fn that's suitable for a conflict.
+     */
+    QString OCSYNC_EXPORT makeConflictFileName(const QString &fn, const QDateTime &dt);
+
+    /**
+     * Returns true when a file is locked. (Windows only)
+     */
+    bool OCSYNC_EXPORT isFileLocked(const QString &fileName);
+
+    /*
+     * This function takes a path and converts it to a UNC representation of the
+     * string. That means that it prepends a \\?\ (unless already UNC) and converts
+     * all slashes to backslashes.
+     *
+     * Note the following:
+     *  - The string must be absolute.
+     *  - it needs to contain a drive character to be a valid UNC
+     *  - A conversion is only done if the path len is larger than 245. Otherwise
+     *    the windows API functions work with the normal "unixoid" representation too.
+     *
+     *  This function allocates memory that must be freed by the caller.
```

Comment:
This is comment seems no longer correct.

Minor: I wonder if this could be two overloads that call a template defined in a cpp file.

Reply:
done